[CI] EntitlementsDeniedNonModularIT class failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #6511 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/y7ihmvuhmrm4e)
- [elasticsearch-periodic-platform-support #6511 / windows-2019_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/u2tglv75hmsf4)
- [elasticsearch-pull-request #58246 / part-1-fips](https://gradle-enterprise.elastic.co/s/nhqblaqej6n76)
- [elasticsearch-pull-request #57909 / part-1-fips](https://gradle-enterprise.elastic.co/s/xlkewzrtt4eba)

**Reproduction Line:**
```
undefined
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.entitlement.qa.EntitlementsDeniedNonModularIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(org.elasticsearch.entitlement.qa.EntitlementsDeniedNonModularIT),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
undefined
```

**Issue Reasons:**
- [8.x] 2 consecutive failures in class org.elasticsearch.entitlement.qa.EntitlementsDeniedNonModularIT
- [8.x] 2 consecutive failures in step part-1-fips
- [8.x] 4 failures in class org.elasticsearch.entitlement.qa.EntitlementsDeniedNonModularIT (2.0% fail rate in 201 executions)
- [8.x] 2 failures in step part-1-fips (100.0% fail rate in 2 executions)
- [8.x] 2 failures in pipeline elasticsearch-pull-request (10.5% fail rate in 19 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

The underlying failure seems to be security not finding a crypto algo:
```
Caused by: java.security.NoSuchAlgorithmException: PBKDF2withHMACSHA512 SecretKeyFactory not available
  at java.base/javax.crypto.SecretKeyFactory.<init>(SecretKeyFactory.java:118)
  at java.base/javax.crypto.SecretKeyFactory.getInstance(SecretKeyFactory.java:164)
  at org.elasticsearch.xcore@9.1.0-SNAPSHOT/org.elasticsearch.xpack.core.XPackSettings.lambda$defaultStoredPasswordHashAlgorithmSetting$3(XPackSettings.java:276)
```

This looks the same as https://github.com/elastic/elasticsearch/issues/122546. But in this case the CI run was _not_ fips. So perhaps entitlements are somehow silently blocking the SecretKeyFactory. I can't see how, though, since we do not instrument that method. A cursory scan of the linked failures seems to show these are all running with JDK 21 on Linux.

--- a/libs/entitlement/src/main/java/org/elasticsearch/entitlement/initialization/EntitlementInitialization.java
+++ b/libs/entitlement/src/main/java/org/elasticsearch/entitlement/initialization/EntitlementInitialization.java
@@ -59,6 +59,7 @@ import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import java.util.stream.StreamSupport;
 
+import static org.elasticsearch.entitlement.runtime.policy.entitlements.FilesEntitlement.Mode.READ;
 import static org.elasticsearch.entitlement.runtime.policy.entitlements.FilesEntitlement.Mode.READ_WRITE;
 
 /**
@@ -149,8 +150,26 @@ public class EntitlementInitialization {
                         new ManageThreadsEntitlement(),
                         new FilesEntitlement(
                             List.of(
-                                FilesEntitlement.FileData.ofPath(EntitlementBootstrap.bootstrapArgs().tempDir(), READ_WRITE),
-                                FilesEntitlement.FileData.ofPath(EntitlementBootstrap.bootstrapArgs().logsDir(), READ_WRITE)
+                                FileData.ofPath(bootstrapArgs.tempDir(), READ_WRITE),
+                                FileData.ofPath(bootstrapArgs.logsDir(), READ_WRITE),
+                                // OS release on Linux
+                                FileData.ofPath(Path.of("/etc/os-release"), READ),
+                                FileData.ofPath(Path.of("/etc/system-release"), READ),
+                                FileData.ofPath(Path.of("/usr/lib/os-release"), READ),
+                                // read max virtual memory areas
+                                FileData.ofPath(Path.of("/proc/sys/vm/max_map_count"), READ),
+                                FileData.ofPath(Path.of("/proc/meminfo"), READ),
+                                // load averages on Linux
+                                FileData.ofPath(Path.of("/proc/loadavg"), READ),
+                                // control group stats on Linux. cgroup v2 stats are in an unpredicable
+                                // location under `/sys/fs/cgroup`, so unfortunately we have to allow
+                                // read access to the entire directory hierarchy.
+                                FileData.ofPath(Path.of("/proc/self/cgroup"), READ),
+                                FileData.ofPath(Path.of("/sys/fs/cgroup/"), READ),
+                                // // io stats on Linux
+                                FileData.ofPath(Path.of("/proc/self/mountinfo"), READ),
+                                FileData.ofPath(Path.of("/proc/diskstats"), READ),
+                                FileData.ofPath(Path.of("/etc/os-release"), READ)
                             )
                         )
                     )

--- a/libs/entitlement/src/main/java/org/elasticsearch/entitlement/initialization/EntitlementInitialization.java
+++ b/libs/entitlement/src/main/java/org/elasticsearch/entitlement/initialization/EntitlementInitialization.java
@@ -168,8 +168,7 @@ public class EntitlementInitialization {
                                 FileData.ofPath(Path.of("/sys/fs/cgroup/"), READ),
                                 // // io stats on Linux
                                 FileData.ofPath(Path.of("/proc/self/mountinfo"), READ),
-                                FileData.ofPath(Path.of("/proc/diskstats"), READ),
-                                FileData.ofPath(Path.of("/etc/os-release"), READ)
+                                FileData.ofPath(Path.of("/proc/diskstats"), READ)
                             )
                         )
                     )

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -338,14 +338,6 @@ tests:
 - class: org.elasticsearch.entitlement.runtime.policy.PolicyParserFailureTests
   method: testEntitlementAbsolutePathWhenRelative
   issue: https://github.com/elastic/elasticsearch/issues/122666
-- class: org.elasticsearch.entitlement.qa.EntitlementsAllowedNonModularIT
-  issue: https://github.com/elastic/elasticsearch/issues/122568
-- class: org.elasticsearch.entitlement.qa.EntitlementsDeniedIT
-  issue: https://github.com/elastic/elasticsearch/issues/122566
-- class: org.elasticsearch.entitlement.qa.EntitlementsDeniedNonModularIT
-  issue: https://github.com/elastic/elasticsearch/issues/122569
-- class: org.elasticsearch.entitlement.qa.EntitlementsAllowedIT
-  issue: https://github.com/elastic/elasticsearch/issues/122680
 - class: org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT
   method: test {yaml=reference/snapshot-restore/apis/get-snapshot-api/line_408}
   issue: https://github.com/elastic/elasticsearch/issues/122681

